src/MaaCore/Assistant.cpp

```diff
@@ -486,6 +492,43 @@ void Assistant::msg_proc()
     }
 }
 
+void Assistant::guard_proc()
+{
+    LogTraceFunction;
+
+    while (true) {
+        std::unique_lock<std::mutex> lock(m_guard_mutex);
+        if (m_thread_exit) {
+            return;
+        }
+
+        if (m_thread_idle || m_tasks_list.empty()) {
+            m_guard_condvar.wait(lock);
+            continue;
+        }
+
+        if (m_ctrler) {
+            const auto& _task = m_tasks_list.front().second;
+            if (auto task = std::dynamic_pointer_cast<StartUpTask>(_task); task) {
+                continue;
+            }
+            const auto& activities = m_ctrler->get_activities();
+            if (!activities) {
+            }
+            else if (activities->find("com.hypergryph.arknights") == std::string::npos) {
+                Log.warn("Assistant::guard_proc | activities died");
+                m_ctrler->start_game("Official");
+                std::this_thread::sleep_for(std::chrono::seconds(120));
+            }
+            else {
+            }
```

Comment:
```suggestion
            if (!activities || activities->empty()) {
                Log.warn("Assistant::guard_proc | activities died");
                m_ctrler->start_game("Official"); // TODO: fix hardcoded game client
                std::this_thread::sleep_for(std::chrono::seconds(120));
            }
```

Reply:
The logic behind this, together with the optimized `grep` is that grep returns nothing in case the client really died, so start it.

If anything is returned, in this case looking at the `grep -i ACTIVITY.*arknights` then the client is running 
_remember that active does not mean running as it could be frozen :( _

Reply:
what would it print? if two different arknights are running.

Reply:
definitely something like this with two arknights instead of only one
```bash
ACTIVITY com.mumu.launcher/.Launcher 625cc04 pid=1417
ACTIVITY com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext 8d8cbd9 pid=2605
```
It can be understood like an `OR` gate as long as there's one arknights running

Reply:
I think we could record the latter and try to start a new one if it crashed

Reply:
Technically the higher PID indicates something that was run later so we can be 90% sure of that. Recording just the package to open that? Good idea.

Reply:
not sure for higher pid.

Some one may open A and B(later and higher pid), then use MAA with A

Reply:
Fair enough. (Still can't understand why people would use the same instance for two clients) when they have to change MAA client. Just use two instances and two MAAs

Reply:
> Technically the higher PID indicates something that was run later so we can be 90% sure of that. Recording just the package to open that? Good idea. After [StartGameTaskPlugin.cpp](https://github.com/MaaAssistantArknights/MaaAssistantArknights/blob/29ca30cb99e27814aeae8a0635b99a4eeee75c0f/src/MaaCore/Task/Miscellaneous/StartGameTaskPlugin.cpp#L28-L51) does its things, we can run this one time, and record the value of the activity, after that it's stored. We continue on running the checker, as soon as the ACTIVITY dies (it's caught by the `if`) we use the stored value. What do you think

Probably no. StartUp is optional, but guard thread is required. so don't connect them

Reply:
I meant that the guard thread needs to wait a bit before starting, in case the StartUp is set to run. Otherwise it would conflict with [StartGameTaskPlugin.cpp](https://github.com/MaaAssistantArknights/MaaAssistantArknights/blob/29ca30cb99e27814aeae8a0635b99a4eeee75c0f/src/MaaCore/Task/Miscellaneous/StartGameTaskPlugin.cpp#L28-L51) itself

Reply:
```
const auto& _task = m_tasks_list.front().second;
if (auto task = std::dynamic_pointer_cast<StartUpTask>(_task); task) {
    continue;
}
```
it's here